arrays/arrays.c

```diff
@@ -83,14 +111,26 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
+  if(index > arr->count){
+    fprintf(stderr, "Index is beyond array bounds.");
+    exit(1);
+  }
 
   // Resize the array if the number of elements is over capacity
+  if(arr->count + 1 > arr->capacity){ // check to see if adding an element would require a resize
+    resize_array(arr);
+  }
 
   // Move every element after the insert index to the right one position
+  for(int i = index; i < arr->count; i++){ 
+    arr->elements[i + 1] = arr->elements[i];
+  }
 
   // Copy the element and add it to the array
+  arr->elements[index] = element;
```

Comment:
You should make a call to strdup() here to "copy" the element in the array but again, not a big deal if you don't. I think it's more of a memory thing than anything else. So it would look like arr->elements[index] = strdup(element). But again, I think functionality is the same with or without it.